codec-http2/src/main/java/io/netty/handler/codec/http2/Http2StreamChannelBootstrap.java

```diff
@@ -0,0 +1,229 @@
+        group0 = group0 == null ? parentChannel.eventLoop() : group0;
+
+        return multiplexCodec.createStreamChannel(parentChannel, group0, handler, options, attributes, streamId);
+    }
+
+    /**
+     * Sets the parent channel that must have the {@link Http2MultiplexCodec} in its pipeline.
+     *
+     * @param parent a registered channel with the {@link Http2MultiplexCodec} in its pipeline. This channel will
+     *               be the {@link Channel#parent()} of all channels created via {@link #connect()}.
+     * @return {@code this}
+     */
+    public Http2StreamChannelBootstrap parentChannel(Channel parent) {
+        channelAndCodec = new ParentChannelAndMultiplexCodec(parent);
+        return this;
+    }
+
+    /**
+     * Sets the channel handler that should be added to the channels's pipeline.
+     *
+     * @param handler   the channel handler to add to the channel's pipeline. The handler must be
+     *                  {@link Sharable}.
+     * @return {@code this}
+     */
+    public Http2StreamChannelBootstrap handler(ChannelHandler handler) {
+        this.handler = checkSharable(checkNotNull(handler, "handler"));
+        return this;
+    }
+
+    /**
+     * Sets the {@link EventLoop} to which channels created with this bootstrap are registered.
+     *
+     * @param group the eventloop or {@code null} if the eventloop of the parent channel should be used.
+     * @return {@code this}
+     */
+    public Http2StreamChannelBootstrap group(EventLoopGroup group) {
+        this.group = group;
+        return this;
+    }
+
+    /**
+     * Specify {@link ChannelOption}s to be set on newly created channels. An option can be removed by specifying a
+     * value of {@code null}.
+     */
+    public <T> Http2StreamChannelBootstrap option(ChannelOption<T> option, T value) {
+        checkNotNull(option, "option must not be null");
+        if (value == null) {
+            options.remove(option);
+        } else {
+            options.put(option, value);
+        }
+        return this;
+    }
+
+    /**
+     * Specify attributes with an initial value to be set on newly created channels. An attribute can be removed by
+     * specifying a value of {@code null}.
+     */
+    public <T> Http2StreamChannelBootstrap attr(AttributeKey<T> key, T value) {
+        checkNotNull(key, "key must not be null");
+        if (value == null) {
+            attributes.remove(key);
+        } else {
+            attributes.put(key, value);
+        }
+        return this;
+    }
+
+    public Channel parentChannel() {
+        ParentChannelAndMultiplexCodec channelAndCodec0 = channelAndCodec;
+        if (channelAndCodec0 != null) {
+            return channelAndCodec0.parentChannel;
+        }
+        return null;
+    }
+
+    public ChannelHandler handler() {
+        return handler;
+    }
+
+    public EventLoopGroup group() {
+        return group;
+    }
+
+    public Map<ChannelOption<?>, Object> options() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove usage of generic wildcard type. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1452)